src/main/java/com/airhacks/afterburner/views/FXMLView.java

```diff
@@ -19,34 +19,38 @@
  * limitations under the License.
  * #L%
  */
+
 import com.airhacks.afterburner.injection.Injector;
+import javafx.application.Platform;
+import javafx.beans.property.ObjectProperty;
+import javafx.beans.property.SimpleObjectProperty;
+import javafx.beans.value.ObservableValue;
+import javafx.collections.ObservableList;
+import javafx.fxml.FXMLLoader;
+import javafx.fxml.Initializable;
+import javafx.scene.Node;
+import javafx.scene.Parent;
+
 import java.io.IOException;
 import java.net.URL;
 import java.util.MissingResourceException;
 import java.util.ResourceBundle;
-import static java.util.ResourceBundle.getBundle;
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.Executor;
 import java.util.concurrent.Executors;
 import java.util.function.Consumer;
 import java.util.function.Function;
 import java.util.function.Supplier;
-import javafx.application.Platform;
-import javafx.beans.property.ObjectProperty;
-import javafx.beans.property.SimpleObjectProperty;
-import javafx.beans.value.ObservableValue;
-import javafx.collections.ObservableList;
-import javafx.fxml.FXMLLoader;
-import javafx.scene.Node;
-import javafx.scene.Parent;
+
+import static java.util.ResourceBundle.getBundle;
 
 /**
  * @author adam-bien.com
  */
-public abstract class FXMLView {
+public abstract class FXMLView<T extends Initializable> {
```

Comment:
Yeah, at least within my projects they're supposed to do so. But you're right, checking the FXMLLoaders implementation: there's no need to implement Initializable. I'll get rid of that boundary.

I'm wondering if there's a use case to entirely not having a controller at all?! Covering this case would make the generic implementation at least questionable. But I'd like to avoid those nasty downcasts.